### PR TITLE
Fix issue with ReLU layer and integer dtype

### LINKED ISSUE #35430: ReLU layer doesn't handle integer dtype
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Debian GNU/Linux 10 (buster)
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): v2.1.0-rc0-47-g064e153 2.1.0-rc1
- Python version: 3.7.6
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: None
- GPU model and memory: NA

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
An exception during `tf.keras.layers.ReLU` construction with integer dtype and `max_value`

**Describe the expected behavior**
A layer is properly constructed and functional

**Code to reproduce the issue**
```
import tensorflow as tf
input = tf.keras.layers.Input(shape=(), name='x', dtype='int64')
y = tf.keras.layers.ReLU(max_value=100, dtype='int64')(input)
```

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/usr/local/lib/python3.7/site-packages/tensorflow_core/python/framework/tensor_util.py in _AssertCompatible(values, dtype)
    323   try:
--> 324     fn(values)
    325   except ValueError as e:

/usr/local/lib/python3.7/site-packages/tensorflow_core/python/framework/tensor_util.py in inner(values)
    262   def inner(values):
--> 263     _ = [_check_failed(v) for v in nest.flatten(values)
    264          if not isinstance(v, expected_types)]

/usr/local/lib/python3.7/site-packages/tensorflow_core/python/framework/tensor_util.py in <listcomp>(.0)
    263     _ = [_check_failed(v) for v in nest.flatten(values)
--> 264          if not isinstance(v, expected_types)]
    265   return inner

/usr/local/lib/python3.7/site-packages/tensorflow_core/python/framework/tensor_util.py in _check_failed(v)
    247   # it is safe to use here.
--> 248   raise ValueError(v)
    249 

ValueError: 0.0

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
<ipython-input-16-2baabe6a9f04> in <module>
      1 input = tf.keras.layers.Input(shape=(), name='x', dtype='int64')
----> 2 y = tf.keras.layers.ReLU(max_value=100, dtype='int64')(input)

/usr/local/lib/python3.7/site-packages/tensorflow_core/python/keras/engine/base_layer.py in __call__(self, inputs, *args, **kwargs)
    771                     not base_layer_utils.is_in_eager_or_tf_function()):
    772                   with auto_control_deps.AutomaticControlDependencies() as acd:
--> 773                     outputs = call_fn(cast_inputs, *args, **kwargs)
    774                     # Wrap Tensors in `outputs` in `tf.identity` to avoid
    775                     # circular dependencies.

/usr/local/lib/python3.7/site-packages/tensorflow_core/python/keras/layers/advanced_activations.py in call(self, inputs)
    317                   alpha=self.negative_slope,
    318                   max_value=self.max_value,
--> 319                   threshold=self.threshold)
    320 
    321   def get_config(self):

/usr/local/lib/python3.7/site-packages/tensorflow_core/python/keras/backend.py in relu(x, alpha, max_value, threshold)
   4373   if clip_max:
   4374     max_value = _constant_to_tensor(max_value, x.dtype.base_dtype)
-> 4375     zero = _constant_to_tensor(0., x.dtype.base_dtype)
   4376     x = clip_ops.clip_by_value(x, zero, max_value)
   4377 

/usr/local/lib/python3.7/site-packages/tensorflow_core/python/keras/backend.py in _constant_to_tensor(x, dtype)
    676       A tensor.
    677   """
--> 678   return constant_op.constant(x, dtype=dtype)
    679 
    680 

/usr/local/lib/python3.7/site-packages/tensorflow_core/python/framework/constant_op.py in constant(value, dtype, shape, name)
    256   """
    257   return _constant_impl(value, dtype, shape, name, verify_shape=False,
--> 258                         allow_broadcast=True)
    259 
    260 

/usr/local/lib/python3.7/site-packages/tensorflow_core/python/framework/constant_op.py in _constant_impl(value, dtype, shape, name, verify_shape, allow_broadcast)
    294       tensor_util.make_tensor_proto(
    295           value, dtype=dtype, shape=shape, verify_shape=verify_shape,
--> 296           allow_broadcast=allow_broadcast))
    297   dtype_value = attr_value_pb2.AttrValue(type=tensor_value.tensor.dtype)
    298   const_tensor = g._create_op_internal(  # pylint: disable=protected-access

/usr/local/lib/python3.7/site-packages/tensorflow_core/python/framework/tensor_util.py in make_tensor_proto(values, dtype, shape, verify_shape, allow_broadcast)
    449       nparray = np.empty(shape, dtype=np_dt)
    450     else:
--> 451       _AssertCompatible(values, dtype)
    452       nparray = np.array(values, dtype=np_dt)
    453       # check to them.

/usr/local/lib/python3.7/site-packages/tensorflow_core/python/framework/tensor_util.py in _AssertCompatible(values, dtype)
    329     else:
    330       raise TypeError("Expected %s, got %s of type '%s' instead." %
--> 331                       (dtype.name, repr(mismatch), type(mismatch).__name__))
    332 
    333 

TypeError: Expected int64, got 0.0 of type 'float' instead.
```

```
tf.constant(0., dtype='int64')
```
fails as well but with different backtrace.

tf_nightly-2.1.0.dev20191226 is affected too.

### PR DESCRIPTION
**Pull request by yongtang**
This fix tries to address the issue raised in #35430 where
ReLU layer + integer dtype causes conversion error.
This PR fixes the isse by replacing float `0.` with `0`.

This fix fixes #35430.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

**Comment by gbaned** [comment]
@yongtang Can you please resolve conflicts? Thanks!

**Comment by yongtang** [comment]
Thanks @qlzh727 @gbaned, the PR has been rebased with conflict resolved. Please take a look.

File: tensorflow/python/keras/engine/base_layer_test.py [hunk 1]
```diff
@@ -630,6 +630,11 @@ def call(self, x, y=None):
     out = self.evaluate(layer(x=x, y=y))
     self.assertAllClose(out, 2 * np.ones((10, 1)))
 
+  def test_relu_layer_with_dtype(self):
```
Comment (qlzh727) [inline tensorflow/python/keras/engine/base_layer_test.py]: Thanks for the change, can you move this to the backend_test.py under test_relu()

**Comment by yongtang** [comment]
Thanks @qlzh727 for the review. The PR has been updated with test case moved to `backend_test.py/test_relu()`. Please take a look and let me know if there are any issues.

**Review by qlzh727 — APPROVED** [review]

**Pull request merged**